lib/src/big_decimal.dart

```diff
@@ -336,6 +336,9 @@ class BigDecimal implements Comparable<BigDecimal> {
     return _add(intVal, -other.intVal, scale, other.scale).intVal.sign;
   }
 
+  @override
+  int get hashCode => 31 * intVal.hashCode + scale;
```

Comment:
This can conflict, right? Should we use `Object.hash()` instead?

Reply:
The implementation is about the same, so it's safe as long as collisions are infrequent.